pkg/operation/botanist/component/kubeapiserver/secrets.go

```diff
@@ -234,17 +234,18 @@ func aesKeyFromSecretData(data map[string][]byte) apiserverconfigv1.Key {
 
 func (k *kubeAPIServer) reconcileSecretServer(ctx context.Context) (*corev1.Secret, error) {
 	var (
-		ipAddresses = append([]net.IP{
-			net.ParseIP("127.0.0.1"),
-		}, k.values.ServerCertificate.ExtraIPAddresses...)
-
-		dnsNames = append([]string{
+		ipAddresses = append([]net.IP{}, k.values.ServerCertificate.ExtraIPAddresses...)
+		dnsNames    = append([]string{
 			v1beta1constants.DeploymentNameKubeAPIServer,
 			fmt.Sprintf("%s.%s", v1beta1constants.DeploymentNameKubeAPIServer, k.namespace),
 			fmt.Sprintf("%s.%s.svc", v1beta1constants.DeploymentNameKubeAPIServer, k.namespace),
 		}, kutil.DNSNamesForService("kubernetes", metav1.NamespaceDefault)...)
 	)
 
+	if k.values.SNI.PodMutatorEnabled || k.values.VPN.HighAvailabilityEnabled {
```

Comment:
```suggestion
	if k.values.SNI.PodMutatorEnabled || (k.values.VPN.Enabled && k.values.VPN.HighAvailabilityEnabled) {
```

Q: Why do we need this for `PodMutatorEnabled`?

Reply:
The pod mutator is a sidecar container in the `kube-apiserver` pod which reuses the same server certificate. The `apiserver-proxy.networking.gardener.cloud` webhook is pointing to `https://127.0.0.1:9443/webhook/pod-apiserver-env` accordingly.